src/pyotp/otp.py

```diff
@@ -20,6 +20,8 @@ def __init__(self, s, digits=6, digest=hashlib.sha1):
         """
         self.digits = digits
         self.digest = digest
+        if type(s) == str:  # Avoids ERROR: test_match_rfc AttributeError: 'int' object has no attribute 'upper'
+            s = ''.join(c for c in s if c.upper() in "ABCDEFGHIJKLMNOPQRSTUVWXYZ234567=")
```

Comment:
I think you missed a 1 in here

Reply:
Please enlighten me, as of which part of the [RFC 4648](https://tools.ietf.org/html/rfc4648#section-6) that says there is a```'1'``` in the the base 32 alphabet.

Reply:
Well that's just odd then.

Reply:
Not really. The core comes down to not mistaking 1 for I because its supposed to be human readable.

Reply:
Yes.  The very same reason ```'0'``` is left out (```'O'``` is used).
https://www.youtube.com/watch?v=deW1GwdAKd0
_bon voyage_